src/frontends/onnx/frontend/src/utils/common.hpp

```diff
@@ -142,6 +142,19 @@ std::shared_ptr<default_opset::Constant> make_failsafe_constant(const ngraph::el
 /// \brief Checks the node's runtime info object and returns true if this node represents
 ///        a dummy failsafe node created instead of an incorrect node found in the original model
 bool is_failsafe_node(const std::shared_ptr<ov::Node>& node);
+
+/// \brief Marks a node with a special entry which indicates that the friendly name has already been set
+///        and any generic mechanism which handles friendly names should skip this node
+void skip_setting_friendly_names(const std::shared_ptr<ov::Node>& node);
+
+/// \brief Checks if this node already has a friendly name set. This information can be used to skip overwriting
+///        the currently set friendly name.
+bool friendly_name_already_set(const std::shared_ptr<ov::Node>& node);
+
+/// \brief Sets friendly names for operators with added bias
+void set_friendly_names(const std::shared_ptr<ov::Node> op,
+                        const std::shared_ptr<ov::Node> added_bias,
```

Comment:
This function doesn't take nor pass ownership of these `ov::Node`'s so copying shared_ptr is obsolete here. Passing these by `const std::shared_ptr&` I consider inaccurate as well, because the matter is `ov::Node` itself and not its ownershiping wrapper.
```suggestion
void set_friendly_names(ov::Node* op,
                        ov::Node* added_bias,
```

Reply:
Note that using pure pointers here implies such a change in declared above `skip_setting_friendly_names` which uses `const shared_ptr&`.

Reply:
I did some changes locally and realized that pure pointer complicates the stuff a bit .. so use passing by const reference.